rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java

```diff
@@ -257,21 +251,41 @@ public boolean isDisposed() {
 
           return receiver
               .next()
-              .doOnSubscribe(
-                  (__) -> {
-                    ByteBuf requestFrame =
-                        RequestResponseFrameFlyweight.encodeReleasingPayload(
-                            allocator, streamId, payload);
-
-                    sendProcessor.onNext(requestFrame);
-                  })
-              .doFinally(
-                  signalType -> {
-                    if (signalType == SignalType.CANCEL) {
-                      sendProcessor.onNext(CancelFrameFlyweight.encode(allocator, streamId));
-                    }
-                    removeStreamReceiver(streamId);
-                  })
+              .transform(
+                  Operators.<Payload, Payload>lift(
+                      (s, actual) ->
+                          new RequestOperator(actual) {
+
+                            int streamId;
```

Comment:
@rstoyanchev maybe it worth moving that field to base class as well to avoid identical on every anonymous class

Reply:
Good idea. That class is for handling requests.

Reply:
fixed